ruchish/feed-reader
# [FR-002]: Render Feed and navigate to Article Detail.

**FR-002:** As a user, I want to see reddit feed for the topic "swift" and on selecting any article it should navigate to article detail showing article title, thumbnail and description.

**Changes:**
- Implemented Reddit API Service.
- Implemented Feed Listing screen and associated view models.
- Implemented Article Detail screen.
- Used **Kingfisher (5.8.2)** Image Library to request remote image.

**Testability:**
- Launch **FeedReader** scheme on **iPhone | iPad** in **Potrait | Landscape** mode.
- Notice it will attempt to load feed from Reddit API and for the topic `swift` internally defined.
- Notice some items in feed will only have article title and some will have title and thumbnail image loaded dynamically on the fly.
- Attempt to select any of the item which will navigate user to **Article Detail** showing Article Title in Navigation Bar, Article Thumbnail if available and Article Description if available.

**TODO:**
- Show Error and Loading View for Feed and Article Detail.
- Create some sort of Image Manager which wraps Kingfisher image library so that we don't need to explicit import Kingfisher library.
